packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -486,94 +509,99 @@ export class RunningSummarizer implements IDisposable {
-            error: ackNack.type === MessageType.SummaryNack ? ackNack.contents.errorMessage : undefined,
-            handle: ackNack.type === MessageType.SummaryAck ? ackNack.contents.handle : undefined,
-        });
+            // Wait for ack/nack
+            const ackNack = await Promise.race([summary.waitAckNack(), pendingTimeoutP]);
+            if (!checkNotTimeout(ackNack)) {
+                return undefined;
+            }
+            this.logger.sendTelemetryEvent({
+                eventName: ackNack.type === MessageType.SummaryAck ? "SummaryAck" : "SummaryNack",
+                category: ackNack.type === MessageType.SummaryAck ? "generic" : "error",
+                timeWaiting: Date.now() - this.heuristics.lastAttempted.summaryTime,
+                sequenceNumber: ackNack.sequenceNumber,
+                summarySequenceNumber: ackNack.contents.summaryProposal.summarySequenceNumber,
+                error: ackNack.type === MessageType.SummaryNack ? ackNack.contents.errorMessage : undefined,
+                handle: ackNack.type === MessageType.SummaryAck ? ackNack.contents.handle : undefined,
+            });
 
-        this.pendingAckTimer.clear();
+            this.pendingAckTimer.clear();
 
-        // Update for success
-        if (ackNack.type === MessageType.SummaryAck) {
-            this.heuristics.ackLastSent();
+            // Update for success
+            if (ackNack.type === MessageType.SummaryAck) {
+                this.heuristics.ackLastSent();
 
-            // since we need a full summary after context reload, we only clear this on ack
-            this.immediateSummary = false;
+                // since we need a full summary after context reload, we only clear this on ack
+                this.immediateSummary = false;
 
-            return true;
-        } else {
-            this.raiseSummarizingError("SummaryNack");
-            return false;
+                return true;
+            } else {
+                this.raiseSummarizingError("SummaryNack");
+                return false;
+            }
+        } finally {
+            this.summarizeTimer.clear();
+            this.pendingAckTimer.clear();
         }
     }
 
-    private async generateSummaryWithLogging(message: string, safe: boolean): Promise<GenerateSummaryData | undefined> {
+    private async generateSummaryWithLogging(
+        message: string,
+        options: Omit<IGenerateSummaryOptions, "summaryLogger">,
+    ): Promise<GenerateSummaryData | undefined> {
+        const { refreshLatestAck, fullTree } = options;
         const summarizingEvent = PerformanceEvent.start(this.logger, {
             eventName: "GenerateSummary",
             message,
             timeSinceLastAttempt: Date.now() - this.heuristics.lastAttempted.summaryTime,
             timeSinceLastSummary: Date.now() - this.heuristics.lastAcked.summaryTime,
-            safe: safe || undefined,
```

Comment:
I'd remove || undefined, I do not think it saves that much in COGSs or bandwidth, but makes it easier to read logs